token/token.go

```diff
@@ -519,14 +519,14 @@ func getNumberStat(str string) *numStat {
 				// binary number
 				continue
 			}
-			if (c == 'e' || c == 'E') && dotFound {
+			if (c == 'e' || c == 'E') && !isExponent && ((isNegative && idx > 2) || (!isNegative && idx > 1)) {
```

Comment:
@martin-sucha Do you think about my changes ?

Reply:
@goccy if it passes the tests, I guess it could work :shrug: Maybe we could replace the parser in `getNumberStat` with regular expression to make it more clear, but I don't see regular expressions used anywhere else in the codebase.